src/Symfony/Component/Messenger/EventListener/SendFailedMessageForRetryListener.php

```diff
@@ -148,6 +143,27 @@ private function shouldRetry(\Throwable $e, Envelope $envelope, RetryStrategyInt
         return $retryStrategy->isRetryable($envelope, $e);
     }
 
+    private function getWaitingTime(Envelope $envelope, \Throwable $throwable, RetryStrategyInterface $retryStrategy): int
+    {
+        $delay = null;
+        if ($throwable instanceof RecoverableExceptionInterface && method_exists($throwable, 'getRetryDelay')) {
+            $delay = $throwable->getRetryDelay();
+        }
+
+        if ($throwable instanceof HandlerFailedException) {
+            foreach ($throwable->getWrappedExceptions() as $nestedException) {
+                if ($nestedException instanceof RecoverableExceptionInterface && method_exists($nestedException, 'getRetryDelay')) {
+                    $delay = $nestedException->getRetryDelay();
+                    if (null !== $delay) {
+                        break;
```

Comment:
instead of stopping on the first one, maybe we should return the minimum one when there are many ?

Reply:
@nicolas-grekas sounds logical :)

Reply:
Not sure about `min` – if the use case is setting the delay based on `Retry-After` header, then it should be `max` :thinking:

Reply:
For my use case it does not really matter, so I'm happy to change to `max` if that's the consensus